modules/secure_storage_connector/main.tf

```diff
@@ -0,0 +1,38 @@
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "group" {
```

Comment:
question: Aren't we already working in the context of a resource group? Should this be a data source instead?

Reply:
@gls4 @jsbroks good point George. Originally I was thinking that because customers would be spinning up this storage account as a new bucket for one team, it made sense to have it be in a separate resource group

now I can see they'd maybe already have a resource group for that team, or could provision one in the UI or another tf module, and use a data source here to get it

do y'all think we should switch to data source?

Reply:
or is there a way to write this so that if they already have a resource group with this name we'd spin the federated credential and storage container up in that resource group? and if they don't have a resource group with that name yet, we'd create a new one?

Reply:
I think we should take existing resource group as an input parameter from the customer if they want to use one. If they don't specify one, we can create ours.